### PR TITLE
Propagate bind and publish port settings and add tests

### PR DESCRIPTION
**Pull request by mxm**
The bind and publish port would always default to _local_ because the settings
are not automatically read in the NetworkService anymore. Instead, they have to
be passed to the NetworkService, otherwise the bind and publish address will
always be _local_.

### CODE REVIEW

File: enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.resolvePublishPort;
+import static java.net.InetAddress.getByName;
+import static java.util.Arrays.asList;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class MqttNettyNetworkConfigTest extends CrateUnitTest {
+
+    @Test
+    public void testPSQLPublishPort() throws Exception {
```
Comment (mfussenegger) [inline enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java]: should probably be called `testMqttPublishPort` instead of `PSQL` - ideally the test method should also give an indication of what aspect exactly it tests.

File: sql/src/test/java/io/crate/protocols/postgres/PostgresNettyPublishPortTest.java [hunk 2]
```diff
@@ -63,6 +73,61 @@ public void testNonUniqueBoundAddress() throws Exception {
...
+            networkService, new AlwaysOKNullAuthentication(), Mockito.mock(SslContextProvider.class));
+        try {
+            psql.doStart();
+        } finally {
+            psql.doStop();
+        }
+
+        // Check override for network.host
+        Settings settingsWithCustomHost = Settings.builder().put("network.host", "cantbindtothis").build();
+        psql = new PostgresNetty(settingsWithCustomHost, Mockito.mock(SQLOperations.class), networkService,
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/protocols/postgres/PostgresNettyPublishPortTest.java]: why are we putting different test cases into the same method if the setup is repeated and quite cheap anyway?

(also in the other tests - would easily be possible to create one test method per test case instead of bundling everything into a single method. It's easier to debug later on in case something breaks and also opens up the possibility to have more meaningful test names)

File: enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+import java.util.Collections;
+
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.MQTT_ENABLED_SETTING;
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.resolvePublishPort;
+import static java.net.InetAddress.getByName;
+import static java.util.Arrays.asList;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class MqttNettyNetworkConfigTest extends CrateUnitTest {
```
Comment (mfussenegger) [inline enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java]: Seems like these tests are not related to the actual change at all - could we add them in a separate commit? I usually assume that tests being added as part of a change cover that change specifically

Comment (mxm) [reply enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java]: What do you mean not related? They test the bind/publish settings which were broken.

File: enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java (hunk 1 above)
Comment (mxm) [reply enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java]: +1

File: sql/src/test/java/io/crate/protocols/postgres/PostgresNettyPublishPortTest.java (hunk 2 above)
Comment (mxm) [reply sql/src/test/java/io/crate/protocols/postgres/PostgresNettyPublishPortTest.java]: The test resembles a logical test entity. Breaking it up is possible but scatters the test logic. If you think it is easier, I'll split it up.

File: enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java (hunk 3 above)
Comment (mxm) [reply enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java]: You mean the first tests? Yes, can add them in a separate commit.

**Comment by mxm** [comment]
Thanks for the comments @mfussenegger. I've made the proposed changes.
